python_modules/dagster/dagster/_core/definitions/decorators/sensor_decorator.py

```diff
@@ -249,9 +254,9 @@ def _wrapped_fn(context):
                     ).format(sensor_name=sensor_name, result=result, type_=type(result))
                 )
 
-        return MultiAssetSensorDefinition(
+        return UnresolvedMultiAssetSensorDefinition(
             name=sensor_name,
-            asset_keys=asset_keys,
+            asset_selection=AssetSelection.keys(*asset_keys),
```

Comment:
probably worth considering if we should be accepting `AssetSelection` instead of `asset_keys` in `@multi_asset_sensor`